src/librustc/mir/interpret/value.rs

```diff
@@ -237,13 +237,20 @@ impl<'tcx, Tag> Scalar<Tag> {
     }
 
     #[inline]
-    pub fn from_uint(i: impl Into<u128>, size: Size) -> Self {
+    pub fn try_from_uint(i: impl Into<u128>, size: Size) -> Option<Self> {
         let i = i.into();
-        assert_eq!(
-            truncate(i, size), i,
+        if truncate(i, size) == i {
+            Some(Scalar::Raw { data: i, size: size.bytes() as u8 })
+        } else {
+            None
+        }
+    }
+
+    #[inline]
+    pub fn from_uint(i: impl Into<u128>, size: Size) -> Self {
+        Self::try_from_uint(i, size).unwrap_or_else(|| {
             "Unsigned value {:#x} does not fit in {} bits", i, size.bits()
```

Comment:
Uh, does this build? I don't think the types match up.^^

Reply:
I rushed it and it does not. Let me push the actually working thing D:

Reply:
There is a small problem here. I had to do `let i = i.into()` here. So `i` can be copied and printed as hex. The other option would be to make `i: impl Into<u128> + LowerHex + Copy`.

Reply:
`impl Into<u128> + LowerHex + Copy` doesn't seem too bad, TBH.^^

Reply:
Is by any chance `let i: u128 = 1u128.into()` a noop?

Reply:
I'd hope so.^^ Why?

Reply:
Because in that case we wouldn't need to add more bounds to `i`. Doing `i.into()` early shouldn't affect `try_from_(u)int` in any way.

Reply:
Hm yeah, I guess that works as well.